backend/open_webui/utils/webhook.py

```
@@ -83,8 +83,9 @@ async def post_webhook(name: str, url: str, message: str, event_data: dict) -> b
             payload = {**event_data}
 
         log.debug(f"payload: {payload}")
-        async with aiohttp.ClientSession() as session:
-            async with session.post(url, json=payload) as response:
+        timeout = aiohttp.ClientTimeout(total=30, connect=10, sock_read=30)
+        async with aiohttp.ClientSession(timeout=timeout) as session:
+            async with session.post(url, json=payload, allow_redirects=False) as response:
                 response.raise_for_status()
                 await response.text()
         return True
```

Ruff (0.13.1)

Consider moving this statement to an `else` block

(TRY300)